### DISCUSSION TITLE
unit:EventPlaysound( soundID, delay )

### DISCUSSION THREAD

**Discussion by cressidagp** [arcemu/arcemu#433] [Ideas]
Hi there... just a suggestion for a distant future.
We already have EventChat(), EventCastSpell() wich are pretty cool. But there are some creatures and bosses who need to play many sound files. Another option is just add a new argument to EventChat()

So instead of do:

function boss_chat1()
end

function boss_chat2()
end

function boss_chat3()
end

We can do this way wich seems more efficient:

function dialogue()
unit:EventPlaysound( soundId1, 4000 )
unit:EventPlaysound( soundId2, 8000 )
unit:EventPlaysound( soundId3, 10000 )
end

**Comment by dfighter1985** [comment]
I'm not very well versed with Arcemu's LUA scripting. Can you show me a real example?
(I find your suggestion reasonable at first sight however)

**Comment by cressidagp** [comment]
Well to be honest im not too much versed either. I just readed all documentation and developers notes that i found in the forum and then looked at the plugin source code to try to get some extra juice. So we have a high chance of this just being another crapy ticket of myself :D (in that case i deeply sorry, just let me now).

Here 4 examples:

[currentWay1_LadyDW.txt](https://github.com/arcemu/arcemu/files/5146317/currentWay1_LadyDW.txt)
[currentWay2_LadyDW.txt](https://github.com/arcemu/arcemu/files/5146318/currentWay2_LadyDW.txt)
[ProposedWay1_ladyDW.txt](https://github.com/arcemu/arcemu/files/5146319/ProposedWay1_ladyDW.txt)
[ProposedWay2_ladyDW.txt](https://github.com/arcemu/arcemu/files/5146320/ProposedWay2_ladyDW.txt)

**Comment by dfighter1985** [comment]
After further considerations now I think that this does make sense, however not for the reason you want it to.
We need to be able to time everything, BUT we shouldn't time sequences this way. 
Let me explain why:
If you time events like this then when the server is busy the timer can run out on all of the timed events, and end up executing the same time.

We need something better.
I will implement this anyways sooner or later.

**Comment by cressidagp** [comment]
Hi there, thanks for taking some time for think on this. Overtime i also realized than bosses usually do emotes and chat when playsound, so i guess EventPlaysound() don't have much sense, plus i didn't notice that issue about timers than you described. 

Anyway I'm happy with the current ALE status and with APE overthere probably it's better to focus on it.

Feel free to close this ticket. And thanks for all @dfighter1985

**Comment by dfighter1985** [comment]
> Hi there, thanks for taking some time for think on this. Overtime i also realized than bosses usually do emotes and chat when playsound, so i guess EventPlaysound() don't have much sense, plus i didn't notice that issue about timers than you described.
> 
> Anyway I'm happy with the current ALE status and with APE overthere probably it's better to focus on it.
> 
> Feel free to close this ticket. And thanks for all @dfighter1985

With APE you'd have the same issue.
The solution I have in mind will have to be implemented in both, but first it needs support in Arcemu itself (the core if you will).

**Comment by dfighter1985** [comment]
@cressidagp you could also try using a "state machine" in AI updates, like I do in this script that I added in commit cd7b8e5b38a0e7bec91f4963eaff9997ea7fdb4e

**Comment by cressidagp** [comment]
Thank you Sir, I will try to script the Lich King encounter in python using that

**Comment by dfighter1985** [comment]
I have converted this ticket to a discussion as a test.
...and it seems to work!
